sequence_processing_pipeline/ConvertJob.py

```diff
@@ -341,33 +337,58 @@ def copy_sequences(self, sample_name, source_project, dest_project,
+            raise ValueError("'copy_all_replicates' is set to 'True' but this "
                              "sample-sheet doesn't contain replicates")
 
         samples = self.info[source_project]['samples']
-        sample_list = []
+
+        # Either we will need to copy all replicates, in which case we need to
+        # match our input to values in the orig_name column, or we want to copy
+        # a single sample, in which case the input needs to be matched to a
+        # value in the sample_name column. The results of our match will be
+        # stored in results and if we are not copying all replicates, there
+        # should only be one sample matched in results.
+        results = []
 
         if copy_all_replicates:
-            for key in samples:
-                sample = samples[key]
-                # assume orig_name is present if treat_as_orig_name is True.
+            for _, sample in samples.items():
+                # assume orig_name is present if copy_all_replicates is True.
                 if sample_name == sample['orig_name']:
-                    sample_list.append(sample)
+                    results.append(sample)
         else:
             # sample_name is a value from the sample_name column. it may or
             # may not have a well-id appended and this sample-sheet may or
             # may not contain replicates, but in either case a single sample
             # either exists or it doesn't.
             if sample_name in self.info[source_project]['samples']:
-                sample_list.append(samples[sample_name])
-
-        if len(sample_list) == 0:
-            # if the sample_list is empty, then sample-name wasn't present in
-            # either the sample_name or orig_name columns.
-            raise ValueError(f"'{sample_name}' is not defined in the project"
-                             f" '{source_project}'")
-
-        for sample in sample_list:
+                results.append(samples[sample_name])
+
+        if len(results) == 0:
+            if copy_all_replicates:
+                # the value of sample_name did not match any value in the
+                # orig_name column. It may match a value in the sample_name
+                # column.
+
+                # if we want to copy all replicates and we mistakenly provide
+                # an input from the sample_name column, we could first look
+                # for a match in the orig_name column and if it fails, look for
+                # an exact match in the sample_name column. If we get an exact
+                # match we can move forward using the input's associated
+                # orig_name. This is not currently implemented however.
+                msg = (f"'{sample_name}' did not match any values in the 'orig"
+                       f"_name' column for project '{source_project}'. Your "
+                       f"value '{sample_name}' have a well-id appended to it")
```

Comment:
Is there a word missing in this error message?  "Your value '<sample_name>' *may* have a well-id appended to it"?